src/sai.rs

```diff
@@ -0,0 +1,71 @@
+//! SAI pad configuration
+
+#![allow(non_snake_case)]
+
+/// An SAI signal
+pub trait Signal: private::Sealed {}
+
+/// A tag that indicates a SAI TX bit clock pad
+pub enum TX_BCLK {}
+
+/// A tag that indicates a SAI TX frame sync pad
+pub enum TX_SYNC {}
+
+/// A tag that indicates a SAI RX bit clock pad
+pub enum RX_BCLK {}
+
+/// A tag that indicates a SAI RX frame sync pad
+pub enum RX_SYNC {}
+
+/// A tag that indicates a SAI MCLK pad
+pub enum MCLK2 {}
+
+impl Signal for TX_BCLK {}
+impl Signal for TX_SYNC {}
+impl Signal for RX_BCLK {}
+impl Signal for RX_SYNC {}
+impl Signal for MCLK2 {}
+
+mod private {
+    pub trait Sealed {}
+    impl Sealed for super::TX_BCLK {}
+    impl Sealed for super::TX_SYNC {}
+    impl Sealed for super::RX_BCLK {}
+    impl Sealed for super::RX_SYNC {}
+    impl Sealed for super::MCLK2 {}
+}
+
+pub trait Pin: super::IOMUX {
+    const ALT: u32;
+    const DAISY: super::Daisy;
+    type Module: super::consts::Unsigned;
+    type Signal: Signal;
+}
+
+pub trait TxDataPin: super::IOMUX {
+    const ALT: u32;
+    type Module: super::consts::Unsigned;
+    type Index: super::consts::Unsigned;
+}
+
+pub trait RxDataPin: super::IOMUX {
+    const ALT: u32;
+    type Module: super::consts::Unsigned;
+    type Index: super::consts::Unsigned;
+}
+
+pub fn prepare<P: Pin>(pin: &mut P) {
+    super::alternate(pin, P::ALT);
+    super::set_sion(pin);
+    unsafe { P::DAISY.write() };
+}
+
+pub fn prepare_txdata<P: TxDataPin>(pin: &mut P) {
+    super::alternate(pin, P::ALT);
+    super::set_sion(pin);
+}
+
+pub fn prepare_rxdata<P: RxDataPin>(pin: &mut P) {
+    super::alternate(pin, P::ALT);
+    super::set_sion(pin);
+}
```

Comment:
Hm, I have not tested SAI input (or even multi-line output), so I'm not really in a position to easily test it. It's not clear to me whether that behaviour would transfer over to the 106x series, since ostensibly the 1010 has no daisy chain registers so the configuration scheme could be different.

Reply:
Let's keep this as-is. Once merged, we'll track our uncertainty / future testing with an issue.